serverlessrepo/publish.py

```diff
@@ -174,6 +179,24 @@ def _is_conflict_exception(e):
     return error_code == 'ConflictException'
 
 
+def _wrap_s3_exception(e):
+    """
+    Wrap S3 access denied exception with a better error message.
+
+    :param e: boto3 exception
+    :type e: ClientError
+    :return: S3PermissionsRequired if S3 access denied or the original exception
+    """
+    error_code = e.response['Error']['Code']
+    message = e.response['Error']['Message']
+
+    if error_code == 'BadRequestException' and "Failed to copy S3 object" in message:
+        match = re.search('bucket=(.+?), key=(.+?)$', message)
```

Comment:
You should code this more defensively. AWS service error messages *shouldn't* change, but you wouldn't want this lib to blow up if it were to change. Can add an extra `if match:` before pulling matching groups out of the result.